src/components/Sidebar.tsx

```diff
@@ -190,61 +235,84 @@ const Sidebar = ({ onClose, ...rest }: SidebarProps) => {
             onClose={onClose} />)}
         </Box>
-        <HStack spacing={{ base: '0', [sidebarBreakpoint]: '6' }} >
-          {/* <IconButton
-            size="lg"
-            variant="ghost"
-            aria-label="open menu"
-            icon={<FiBell />}
-            /> */}
-          <Menu>
-            <MenuButton
-              marginX={4}
-              marginBottom={4}
-              paddingLeft={4}
-              py={2}
-              transition="all 0.3s"
-              _focus={{ boxShadow: 'none' }}>
-              <HStack>
-                <Avatar
-                  size={'sm'}
-                  bg="brand.a"
-                  color="white"
-                  name={formatUserName(me.name)} />
-                <Text
-                  display={{ base: 'flex', [sidebarBreakpoint]: 'flex' }}
-                  fontSize="sm">
-                  {formatUserName(me.name)} </Text>
-                <Box display={{ base: 'flex', [sidebarBreakpoint]: 'flex' }}><FiChevronRight /></Box>
-              </HStack>
-            </MenuButton>
-            <MenuList
-              bg={useColorModeValue('white', 'gray.900')}
-              borderColor={useColorModeValue('gray.200', 'gray.700')}>
-              <MenuItem as={NextLink} href='/profile'>个人信息</MenuItem>
-              <MenuDivider />
-              <MenuItem as={NextLink} href='/who-can-see-my-data'>
-                <LockIcon marginRight={1} />谁能看到我的数据
-              </MenuItem>
-              <MenuDivider />
-              <MenuItem onClick={() => signOut()}>退出登录</MenuItem>
-            </MenuList>
-          </Menu>
-        </HStack>
+        <Box>
+          <DropdownMenu 
+            title={userName} 
+            icon={<Avatar size={'sm'} bg="brand.a" color="white" name={userName} 
+              />}
+            menuItems={userDropdownMenuItems}
+          />
+        </Box>
       </Flex>
     </Box >
   );
 };
 
 export default Sidebar;
 
+function DropdownMenu({ title, icon, menuItems, rolesRequired } : {
+  title: string,
+  icon: React.ReactNode,
+  menuItems: DropdownMenuItem[],
+  rolesRequired?: Role | Role[],
+}) {
+  const [user] = useUserContext();
+  const backgroundColor = useColorModeValue(bgColorModeValues[0], 
+    bgColorModeValues[1]);
+  const borderColor = useColorModeValue(bdColorModeValues[0], 
+    bdColorModeValues[1]);
+  const filteredItems = rolesRequired ? 
+    managerDropdownMenuItems.filter(item => isPermitted(user.roles, item.roles))
+    :
+    menuItems;
+
+  if (filteredItems.length === 0) {
+    return <></>;
+  }
+  return <Flex paddingY={sidebarItemPaddingY}>
+    <Menu placement='right-start'>
+    <DropdownMenuButton title={title} icon={icon}/>
+    <MenuList bg={backgroundColor} borderColor={borderColor}>
+      {filteredItems.map((item, index) =>
+        <MenuItem 
+          key={index}
+          as={typeof item.action === 'string' ? NextLink : undefined}
```

Comment:
define an variable: `const isUrl = typeof item.action === 'string'` and use this variable in line 291 through 293.